### ISSUE TITLE
Change log level at runtime

### ISSUE THREAD

**Issue by dannykopping** [grafana/loki#6805]
**Is your feature request related to a problem? Please describe.**
As a Loki operator, I want to be able to change the log level of a Loki process at runtime - particularly from `info` to `debug` - so that I can access debugging information during an incident.

**Describe the solution you'd like**
An HTTP API which configures the log level of the global logger.

**Describe alternatives you've considered**
Run with `debug` level all the time, but this produces quite a large volume of data.

**Additional context**
N/A

**Comment by gyanesh-m** [comment]
Hi @dannykopping ! I would like to pick this up.

**Comment by dannykopping** [comment]
> Hi @dannykopping ! I would like to pick this up.

Go for it :+1:

**Comment by gyanesh-m** [comment]
one clarification, this assumes that the update will be done for one instance only and not for distributed instances?

**Comment by dannykopping** [comment]
Correct: I think the way I would personally like to use this is to create a port-forward to the pod (if we're in k8s) and call an API to change the log level.

**Comment by patsevanton** [comment]
Please add the ability to change the log level from info to error for ingester and querier in loki-distributed.
